keps/1093-multi-level-cluster-queues/README.md

```diff
@@ -0,0 +1,151 @@
+any review.
+-->
+
+<!--
+a table of contents is helpful for quickly jumping to sections of a kep and for
+highlighting any additional information provided beyond the standard KEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [PlanA](#plana)
+  - [Validation](#validation)
+    - [ClusterQueue](#clusterqueue)
+    - [Hierachy](#hierachy)
+    - [ResourceFlavor](#resourceflavor)
+    - [localqueue](#localqueue)
+  - [Schedule Behavior](#schedule-behavior)
+  - [API](#api)
+- [Implementation](#implementation)
+- [Testing Plan](#testing-plan)
+  - [NonRegression](#nonregression)
+  - [Unit Tests](#unit-tests)
+  - [Integration tests](#integration-tests)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+This proposal allow cluster admins to define a multi-level hierachy for cluster queues. Multi-level of cluster queues allow admins and users to define different dequeue policies for different node. This will allow Kueue to manage more levels of resources.
+## Motivation
+Systems like Yarn allow creating a hierarchy of fair sharing, which allows modeling deeper organizational structures with fair-sharing.
+Kueue currently supports three organizational levels: Cohort (models a business unit), ClusterQueue (models divisions within a business unit), namespace (models teams within a division). However fair-sharing is only supported at one level, within a cohort. This is not convinent if there are more than one level in an organization or if some users want to manage the resource consumption of his/her own jobs.
+### Goals
+
+- Allow admins to define multi-level hierachy of cluster queues.
+- Allow users to define weights for different job of him/her. 
+### Non-Goals
+
+-  Cohort will not be deprecated because Kueue is forward campatible.
+## Proposal
+We will extend cluster queues to allow a cluster queue to be the parent of another cluster queue. And allow admins to define weights for all these cluster queues.
+### PlanA
+We propose to add a `.Spec.Children`field to the ClusterQueue CRD. `.Spec.Children`field is a list of `string` to define the children CQs, we will not check if the child CQ exist or not. Noted that `.Spec.Children` cannot exist with `.Spec.Cohort` concurrently.
+We propose to add a crd named `ClusterQueueTree` to contains `.status.trees[]` to display current trees' state. This crd will only show the hierachy, if admins want to query the resource consumption of cluster queue, they need to query through `ClusterQueue` crd.
+```go
+type Policy string
+var FIFO Policy = "fifo"
+var Fair Policy = "fair"
+var Priority Policy = "priority"
+var Capacity Policy = "Capacity"
+type ClusterQueueSpec struct {
+	...
+    Policy Policy
```

Comment:
What is this policy exactly for?

Reply:
I propose to add policy to control the admission sequence for workloads belong to different cluster queues.